libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2660,6 +2663,21 @@ map<ContractDefinition const*, vector<ASTPointer<frontend::Expression>>> SMTEnco
 	return baseArgs;
 }
 
+TypePointer SMTEncoder::isConstant(Expression const& _expr)
+{
+	if (
+		auto type = _expr.annotation().type;
+		type->category() == Type::Category::RationalNumber
+	)
+		return type;
+
+	/// We ignore the errors found by ConstantEvaluator since _expr
+	/// doesn't necessarily have to be a const expression.
```

Comment:
What do you mean by this comment?

Reply:
ConstantEvaluator takes a reference to ErrorReporter where it writes problems it finds, when the given expression is supposed to be const evaluated. Here we don't know for sure that it is supposed to be const evaluated, to we don't really care about the errors that ConstantEvaluator finds. So we create a local ErrorReporter that we discard

Reply:
Yeah but what does `.evaluate` returns in case of error?

Reply:
I assume it returns `nullptr`, so I'd just rephrase the comment as follows:
> We ignore any warnings emitted by the constant evaluator, as it will return nullptr in case of failure.

Reply:
And this comment doesn't need to be doxygen `///`.

Reply:
Done